### PR TITLE
feat: Add tables for database diagram

### PR DESCRIPTION
**Pull request by rotimiazeez**
# Create a database based on a diagram🎉 

> In this project we get a database diagram, interpret it, and implement the SQL needed to build the database💪🏽 .

Additional description of the project and its features ➕:

- clinic_diagram
 
## Built With ❤️ 

- SQL

## Deployed Preview 🚀 
[Deployed Preview](Coming soon)

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    id INT GENERATED ALWAYS AS IDENTITY,
+    unit_price DECIMAL,  
+    quantity INT,
+    total_price DECIMAL, 
+    invoice_id INT,    
+    treatment_id INT,            
+    FOREIGN KEY (invoice_id) REFERENCES invoices(id),
+    FOREIGN KEY (treatment_id) REFERENCES treatments(id),    
+    PRIMARY KEY (id)
+);
```
Comment (de-scholar) [inline schema_based_on_diagram.sql]: - [x] We are missing two important notes:
  - [x] All FOREIGN KEYS have indexes to improve the system's performance
  - [x] All "join table" for the many-to-many relastionships should be created, even if they are not showed on the diagram. Therefore you should create a table between `medical_histories` and `treatments`

**Review by de-scholar — CHANGES_REQUESTED** [review]
Hi @rotimiazeez and @Hamzaoutdoors , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (Hamzaoutdoors) [reply schema_based_on_diagram.sql]: Done Thanks for your feedback ^^

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -54,3 +54,6 @@ CREATE TABLE medical_history_treatments (
     PRIMARY KEY (medical_history_id, treatment_id)
 );
 
+CREATE INDEX ON patients (patient_id);
+CREATE INDEX ON medical_histories (medical_history_id);
+CREATE INDEX ON treatments (treatment_id);
```
Comment (de-scholar) [inline schema_based_on_diagram.sql:59]: :+1: Well done :heavy_check_mark:

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -44,4 +44,13 @@ CREATE TABLE invoice_items(
...
\ No newline at end of file
+);
+
+CREATE TABLE medical_history_treatments (
+    medical_history_id  INT,
+    treatment_id     INT,
+    FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id),
+    FOREIGN KEY (treatment_id) REFERENCES treatments (id),
+    PRIMARY KEY (medical_history_id, treatment_id)
+);
```
Comment (de-scholar) [inline schema_based_on_diagram.sql:55]: :+1: Well done :heavy_check_mark:

**Review by de-scholar — APPROVED** [review]
Hi @rotimiazeez and @Hamzaoutdoors , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**